### PR TITLE
feat: allow setting webview bg color, closes #197

### LINKED ISSUE #197: Allow setting the browser's background color during attributes
**Is your feature request related to a problem? Please describe.**
Allow the user to set a background color for the webview window to render while loading content. This is typically the step between the os window initializing the webview, and the webview initializing the application.
https://github.com/tauri-apps/tauri/issues/1564 ([my comment](https://github.com/tauri-apps/tauri/issues/1564#issuecomment-825871338))

**Describe the solution you'd like**
Having a `background_color` field on `wry::Attributes` (I think that is the proper place for it).
I don't know which format we should accept the color in...
* hex string?
* rgb struct?
* rgb tuple?
* other?

**Describe alternatives you've considered**
none yet

**Would you assign yourself to implement this feature?**
- [ ] Yes
- [x] No (I am not familiar with the macOS code and don't have access to an macOS machine)

**Additional context**
The linux part seems easy enough once a format for accepting the color is decided on.  Some pseudo-code to replace:
https://github.com/tauri-apps/wry/blob/765fe5ae413a1c13ad99802b49f3af859a2445d5/src/webview/linux/mod.rs#L138-L146
with
```rust
let alpha = (!transparent) as i8 as f64;
webview.set_background_color(&RGBA {
  red: bg_color.red,
  green: bg_color.green,
  blue: bg_color.blue,
  alpha,
});
```

assuming `bg_color` is some sort of struct with red, blue, green fields of `f64`.

I assume webview2 and macOS has a way of setting the background color but I'm not familiar with them yet

### PR DESCRIPTION
**Pull request by amrbashir**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [ ] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/wry/blob/dev/.changes/readme.md).
- [x] I have added a convincing reason for adding this feature, if necessary

## TODOs:
- [x] android
- [ ] macOS

### CODE REVIEW

File: src/webview/mod.rs [hunk 1]
```diff
@@ -61,8 +61,24 @@ pub struct WebViewAttributes {
...
+  /// ## Platform-specific:
+  ///
+  /// **Android**: Not implemented.
+  /// **Windows 7**: Not supported.
   pub transparent: bool,
+  /// Specify the webview background color. This will be ignored if `transparent` is set to `true`.
+  ///
+  /// The color uses the RGBA format.
+  ///
+  /// ## Platfrom-specific:
```
Comment (lucasfernog) [inline src/webview/mod.rs]:
```suggestion
  /// ## Platform-specific:
```

File: src/webview/mod.rs [hunk 2]
```diff
@@ -61,8 +61,24 @@ pub struct WebViewAttributes {
...
+  ///
+  /// - **Android**: Not implemented..
```
Comment (lucasfernog) [inline src/webview/mod.rs]:
```suggestion
  /// - **Android**: Not implemented.
```

File: src/webview/mod.rs [hunk 3]
```diff
@@ -61,8 +61,24 @@ pub struct WebViewAttributes {
...
+  /// - **Windows**:
+  ///   - on Windows 7, transparency is not supported and the alpha value will be ignored.
```
Comment (lucasfernog) [inline src/webview/mod.rs]:
```suggestion
  ///   - On Windows 7, transparency is not supported and the alpha value will be ignored.
```

File: src/webview/mod.rs [hunk 4]
```diff
@@ -61,8 +61,24 @@ pub struct WebViewAttributes {
...
+  ///   - on Windows higher than 7: translucent colors are not supported so any alpha value other than `0` will be replaced by `255`
```
Comment (lucasfernog) [inline src/webview/mod.rs]:
```suggestion
  ///   - On Windows higher than 7: translucent colors are not supported so any alpha value other than `0` will be replaced by `255`
```

File: src/webview/mod.rs [hunk 5]
```diff
@@ -535,6 +572,20 @@ impl WebView {
...
+
+  /// Specify the webview background color.
+  ///
+  /// The color uses the RGBA format.
+  ///
+  /// ## Platfrom-specific:
+  ///
+  /// - **Android**: Not implemented.
+  /// - **Windows**:
+  ///   - on Windows 7, transparency is not supported and the alpha value will be ignored.
```
Comment (lucasfernog) [inline src/webview/mod.rs]:
```suggestion
  ///   - On Windows 7, transparency is not supported and the alpha value will be ignored.
```

File: src/webview/mod.rs [hunk 6]
```diff
@@ -535,6 +572,20 @@ impl WebView {
...
+  ///   - on Windows higher than 7: translucent colors are not supported so any alpha value other than `0` will be replaced by `255`
```
Comment (lucasfernog) [inline src/webview/mod.rs]:
```suggestion
  ///   - On Windows higher than 7: translucent colors are not supported so any alpha value other than `0` will be replaced by `255`
```

**Comment by lucasfernog** [comment]
Works on Linux and Windows. Does not work on macOS though :(

EDIT: Ahh I just saw you removed it :P

**Comment by amrbashir** [comment]
yeah I removed the macOS implementation since I can't test it and from my research it needs a lot more work

File: src/webview/android/main_pipe.rs [hunk 7]
```diff
@@ -136,8 +154,35 @@ fn find_my_class<'a>(
...
+  env.call_method(webview, "setBackgroundColor", "(I)V", &[color])?;
+  Ok(())
+}
+
 #[derive(Debug)]
 pub enum WebViewMessage {
-  CreateWebView(String, bool),
+  CreateWebView {
+    url: String,
+    devtools: bool,
```
Comment (lucasfernog) [inline src/webview/android/main_pipe.rs]: we should take the entire webviewattributes struct here

Comment (amrbashir) [reply src/webview/android/main_pipe.rs]: I don't think we can, there is bunch of handlers in WebViewAttributes that aren't send+sync

Comment (lucasfernog) [reply src/webview/android/main_pipe.rs]: well just create a dedicated struct then, otherwise this enum value will soon have 10 fields

**Pull request merged**